Make RGW doc examples more generic (#623)
openstack-k8s-operators/data-plane-adoption: docs_user/modules/proc_migrating-the-rgw-backends.adoc

----

ifeval::["{build}" != "downstream"]
. During the overcloud deployment, a `cephadm`-compatible spec is generated in
  `/home/ceph-admin/specs/rgw`. Find and patch the RGW spec, specify the right placement by using labels,

>   and change the RGW back-end port to `8090` to avoid conflicts with the Ceph Ingress daemon front-end port.

Author: Done

> . During the overcloud deployment, a cephadm compatible spec is generated in
> `/home/ceph-admin/specs/rgw`. Find and patch the RGW spec, specifying the right placement using the labels
> approach, and change the rgw backend port to 8090 to avoid conflicts with the Ceph Ingress daemon frontend port.

Reviewer: ```suggestion
. During the overcloud deployment, a `cephadm`-compatible spec is generated in
`/home/ceph-admin/specs/rgw`. Find and patch the RGW spec, specify the right placement by using labels, and change the RGW back-end port to `8090` to avoid conflicts with the Ceph Ingress daemon front-end port.
```

Specify the right placement of the nodes? If yes, we should add that. E.g., "...specify the right placement of the nodes by using labels..."

Author: Yes, it works for me, I'm pushing a new change for that incorporating your suggestion.

  rgw_zone: default

> ----
> 
> The example above assumes 172.17.3.0/24 is the `storage` network.

Reviewer: ```suggestion
----
+
This example assumes that `172.17.3.0/24` is the `storage` network.
```

Author: Done

. In the `placement` section, ensure that the `label` and `rgw_frontend_port` values are set:
+
----
---
networks:
- 172.17.3.0/24<1>
placement:
  label: rgw <2>
service_id: rgw
service_name: rgw.rgw
service_type: rgw
spec:
  rgw_frontend_port: 8090 <3>
  rgw_realm: default
  rgw_zone: default
----
+

> <1> Add in the network section the storage network where the RGW backends are supposed to be deployed.

Reviewer: ```suggestion
<1> Add the storage network where the RGW back ends are deployed.
```

Author: Done

rgw.rgw.cephstorage-2.krycit  cephstorage-2  172.17.3.81:8090  running (13s)
----

. Ensure that the new RGW back ends are reachable on the new ports, so you can enable an IngressDaemon on port `8080` later. Log in to each {CephCluster} node that includes RGW and add the `iptables` rule to allow connections to both 8080 and 8090 ports in the {CephCluster} nodes:
+
----
$ iptables -I INPUT -p tcp -m tcp --dport 8080 -m conntrack --ctstate NEW -m comment --comment "ceph rgw ingress" -j ACCEPT
$ iptables -I INPUT -p tcp -m tcp --dport 8090 -m conntrack --ctstate NEW -m comment --comment "ceph rgw backends" -j ACCEPT
----

. From a Controller node, such as `controller-0`, try to reach the RGW back ends:
+
----
$ curl http://cephstorage-0.storage:8090;
----
+
You should observe the following output:
+
----
<?xml version="1.0" encoding="UTF-8"?><ListAllMyBucketsResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Owner><ID>anonymous</ID><DisplayName></DisplayName></Owner><Buckets></Buckets></ListAllMyBucketsResult>

> ----
> 
> Repeat the verification for each node where a RGW daemon is deployed.

Reviewer: ```suggestion
----
+
Repeat the verification for each node where a RGW daemon is deployed.
```

You need the plus sign to align the text under the code block.

Author: Done (thank you)

. If you migrated RGW back ends to the {Ceph} nodes, there is no `internalAPI` network, except in the case of HCI nodes. You must reconfigure the RGW keystone endpoint to point to the external network that you propagated:
+
----
[ceph: root@controller-0 /]# ceph config dump | grep keystone
global   basic rgw_keystone_url  http://172.16.1.111:5000

> [ceph: root@controller-0 /]# ceph config set global rgw_keystone_url http://<keystone_endpoint>:5000<1>
> ----
> +
> <1> Replace `keystone_endpoint` with the keystone internal endpoint of the service deployed in the `OpenStackControlPlane` during the Keystone adoption procedure.
> 

Reviewer: ```suggestion
[ceph: root@controller-0 /]# ceph config set global rgw_keystone_url http://<keystone_endpoint>:5000
----
+
* Replace `<keystone_endpoint>` with the {identity_service_first_ref} internal endpoint of the service that is deployed in the `OpenStackControlPlane` CR when you adopt the {identity_service}.  For more information, see xref: adopting-the-identity-service_adopt-control-plane[Adopting the {identity_service}].

```

Author: Done
